### PR TITLE
Add test to cover Quarkus configuration properties

### PR DESCRIPTION
**Pull request by kshpak**
This test covers most of the config properties for Quarkus described in Configuring you Quarkus application guide
https://access.redhat.com/documentation/en-us/red_hat_build_of_quarkus/1.7/html-single/configuring_your_quarkus_applications/index

resources-config.json file should be there according to https://quarkus.io/guides/writing-native-applications-tips#supporting-native-in-your-application (without it, the errors are thrown on the following line
`InputStream in = CustomConfigSource.class.getResourceAsStream(PROPERTIES_FILE);`
)

Module has reach README file

### CODE REVIEW

File: 022-quarkus-properties-config-all/src/main/java/org/acme/GreetingResource.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import org.acme.converter.KrustyEmail;
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
+import javax.inject.Inject;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Path("/hello")
+public class GreetingResource{
```
Comment (rsvoboda) [inline 022-quarkus-properties-config-all/src/main/java/org/acme/GreetingResource.java:19]:
```suggestion
public class GreetingResource {
```
the same is on few other places

File: 022-quarkus-properties-config-all/pom.xml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      <artifactId>quarkus-config-yaml</artifactId>
+    </dependency>
+  </dependencies>
+
+  <profiles>
+    <profile>
+      <id>native</id>
+      <properties>
+        <quarkus.package.type>native</quarkus.package.type>
+        <quarkus.native.additional-build-args>-H:ResourceConfigurationFiles=resources-config.json</quarkus.native.additional-build-args>
```
Comment (rsvoboda) [inline 022-quarkus-properties-config-all/pom.xml:29]: No hint is needed for JVM mode, just for native mode ?

**Review by rsvoboda — CHANGES_REQUESTED** [review]
More general comments:
 - use quarkus qe package name 
 - can be this test coverage merged with existing module - e.g. 001-quarkus-getting-started-with-jaxrs?  Motivation is to save time on native image compilation.
    - 001-quarkus-getting-started-with-jaxrs => 001-quarkus-jaxrs-and-config-properties

File: 022-quarkus-properties-config-all/src/main/java/org/acme/GreetingResource.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    @Inject
+    Config config;
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public String welcomeMessage(){
+        return config.getValue("welcome.message", String.class);
+    } // declared in .env
+
+
```
Comment (Sgitario) [inline 022-quarkus-properties-config-all/src/main/java/org/acme/GreetingResource.java:30]: Double empty line here.

File: 022-quarkus-properties-config-all/src/main/java/org/acme/GreetingResource.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    @GET
+    @Path("/antagonist/wife")
+    @Produces(MediaType.TEXT_PLAIN)
+    public String helloAntagonistWife() {
+        return antagonist.wife.name + " says: " + antagonist.wife.message;
+    }
+
+
+    @ConfigProperty(name = "restaurant.employees")
+    List<KrustyEmail> employeesEmails;
```
Comment (Sgitario) [inline 022-quarkus-properties-config-all/src/main/java/org/acme/GreetingResource.java:68]: Fields should be defined before methods. Same with the other fields.

File: 022-quarkus-properties-config-all/pom.xml (hunk 2 above)
Comment (kshpak) [reply 022-quarkus-properties-config-all/pom.xml:29]: right, just for native mode. without it tests still pass, but there are a bunch of errors printed before, which slowdown native execution. errors related to loading additional config source

**Comment by kshpak** [comment]
All comments reflected in changes. I would prefer to leave it as a separate module

File: 022-quarkus-properties-config-all/src/test/java/quarkus/qe/GreetingResourceTest.java [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+import io.quarkus.test.junit.QuarkusTest;
+import org.junit.jupiter.api.Test;
+
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.CoreMatchers.is;
+
+@QuarkusTest
+public class GreetingResourceTest {
```
Comment (rsvoboda) [inline 022-quarkus-properties-config-all/src/test/java/quarkus/qe/GreetingResourceTest.java:11]: I would use something like `VariousConfigurationSourcesTest`

File: 022-quarkus-properties-config-all/src/test/java/quarkus/qe/GreetingResourceTest.java [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+    public void testEnvFile() {
+        given()
+          .when().get("/hello")
+          .then()
+             .statusCode(200)
+             .body(is("Welcome message from .env file"));
+    }
+
+    @Test
+    public void testProtagonistEndpoint() {
```
Comment (rsvoboda) [inline 022-quarkus-properties-config-all/src/test/java/quarkus/qe/GreetingResourceTest.java:23]: Indicate what is the source of the configuration (application properties, yaml, resources-config.json etc) in the test name
Do not use the endpoint name, that doesn't hint the problematic area if the test fails

**Review by rsvoboda — CHANGES_REQUESTED** [review]

File: 022-quarkus-properties-config-all/src/test/java/quarkus/qe/GreetingResourceTest.java (hunk 6 above)
Comment (kshpak) [reply 022-quarkus-properties-config-all/src/test/java/quarkus/qe/GreetingResourceTest.java]: Good point. Changed

**Review by Sgitario — APPROVED** [review]

**Review by rsvoboda — APPROVED** [review]

**Comment by rsvoboda** [comment]
waiting for ci ...

**Pull request merged**